### PR TITLE
PROC-876: V3 README Update

### PR DESCRIPTION
**Pull request by jordanross-wf**
## Ultimate problem:
- Needed to update our readme's for new v3 changes

## How it was fixed:
- Update MIGRATION.md
- Update README.md
- Update the comment on the public `refMap` getter.

## Testing suggestions:
N/A

## Potential areas of regression:
N/A

---

> __FYA:__ @michaelcarter-wf

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by semveraudit-wf** [comment]
### Public API Changes

 No changes to the public API found for commit 6f579781517c0f6662bf46668d44fade180f358b

Showing results for 6f579781517c0f6662bf46668d44fade180f358b
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
> Reported by the dart semver audit client 2.2.1
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/json_schema/6f579781517c0f6662bf46668d44fade180f358b).

**Review by michaelcarter-wf — APPROVED** [review]
+1

File: MIGRATION.md [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
+- `JsonSchema.resolvePath` now takes a URI object as a parameter instead of a string.
+- The `refProvider` parameter in the constructors is now a class you can configure to provide resolved `JsonSchema`s or valid JSON objects, synchronously or asynchronously.
+  - 4 Static methods can be used for easy construction:
+    - `RefProvider.syncSchema`
+    - `RefProvider.syncJson`
+    - `RefProvider.asyncSchema`
+    - `RefProvider.asyncJson`
+- The `_refMap` member used to contain references to all sub-schemas in the schema and remote refs, but now only contains remote referenced schemas and sub-schemas with unique `$id`s.
+  - The public getter `refMap` is marked as deprecated due to the fact it shouldn't need to be used externally, but `resolvePath` should be used to fetch schemas from a given URI.
+- Draft 7 is now supported!
```
Comment (michaelcarter-wf) [inline MIGRATION.md:13]: Draft 7 is now also the default if you don't specify, so that's another breaking behavioral change

**Review by michaelcarter-wf — CHANGES_REQUESTED** [review]

**Review by michaelcarter-wf — DISMISSED** [review]
+1

**Comment by michaelcarter-wf** [comment]
@Workiva/release-management-p

**Review by michaelcarter-wf — APPROVED** [review]
+1

**Comment by michaelcarter-wf** [comment]
@Workiva/release-management-p

**Comment by michaelcarter-wf** [comment]
@Workiva/release-management-p

**Comment by michaelcarter-wf** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**